File: tyama_icpc2007dA.swift
//usr/bin/env swift $0 $@;exit
var n=Int(readLine()!)!
while n>0 {
	var ma=0
	var mi=1000
	var sum=0
	for i in 0..<n {
		var s=Int(readLine()!)!
		if ma<s {ma=s}
		if mi>s {mi=s}
		sum+=s
	}
	print((sum-ma-mi)/(n-2))
	n=Int(readLine()!)!
}

File: tyama_icpc2007dA.nim
#[
/usr/bin/env nim c --nimcache:$HOME/.nimcache --opt:speed --run $0 $@;exit
#]#
import strutils
while true:
 var n=parseInt(readLine(stdin))
 if n==0: break
 var s=0
 var mi=1000
 var ma=0
 for i in 1..n:
  var x=parseInt(readLine(stdin))
  if x<mi: mi=x
  if x>ma: ma=x
  s=s+x
 echo((s-ma-mi) div (n-2))
